src/seq.rs

```diff
@@ -0,0 +1,119 @@
+    slope_length: usize,
+
+    /// The position _relative to the last chunk boundary_.
+    /// Specifically, this resets to zero after a chunk is emitted.
+    /// We need this to track if we're done filling our window between calls to `find_boundary`,
+    /// since we're fed blocks of data which may be smaller than the window.
+    pos: usize,
+}
+
+impl SeqChunkerState {
+    fn reset(&mut self) {
+        self.increment_run_length = 1;
+        self.previous_value = None;
+        self.slope_length = 1;
+        self.pos = 0;
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct SeqChunker {
+    /// The length of the current incrementing run.
+    seq_length: usize,
+
+    /// Length of opposing slope required to trigger a chunk boundary.
+    skip_trigger: usize,
+
+    /// The number of bytes to skip ahead when the skip trigger is met.
+    skip_size: usize,
+
+    /// The current state of the chunker.
+    state: SeqChunkerState,
+}
+
+impl SeqChunker {
+    pub fn new(seq_length: usize, skip_trigger: usize, skip_size: usize) -> Self {
+        SeqChunker {
+            seq_length,
+            skip_trigger,
+            skip_size,
+            state: Default::default(),
+        }
+    }
+}
+
+impl ChunkerImpl for SeqChunker {
+    /// Searches for a chunk boundary in the provided data block.
+    /// If a boundary is found that is not the very end of the data block,
+    /// its index is returned within an Option.
+    fn find_boundary(&mut self, data: &[u8]) -> Option<usize> {
+        let mut i = 0;
+        
+        while i < data.len() {
+            let b = data[i];
+            
+            if let Some(previous_value) = self.state.previous_value {
+                // Compare current and previous byte
+                let cmp_result = b - previous_value;
+                
+                // Low Entropy Absorption - skip identical bytes
+                if cmp_result == 0 {
+                    i += 1;
+                    self.state.previous_value = Some(b);
+                    continue;
+                }
+                
+                let cmp_sign = cmp_result.is_negative();
+                
+                // Increment opposing slope count when current < previous (cmp_sign is true)
+                self.state.slope_length += cmp_sign as usize;
+                
+                // Reset and increment sequence length when current > previous (cmp_sign is false)
+                self.state.increment_run_length = (self.state.increment_run_length * (!cmp_sign as usize)) + (!cmp_sign as usize);
+                
+                // Check if we've found a boundary
+                if self.state.increment_run_length == self.seq_length {
+                    return Some(i);
+                }
+                
+                // Check if we need to skip ahead
+                if self.state.slope_length == self.skip_trigger {
+                    i += self.skip_size;
+                    self.state.reset();
```

Comment:
Strictly speaking, our algorithm isn't exactly what they are implementing, but I don't think it's clear from the paper (maybe double check).

They skip ahead and compare `data[i] - data[i-1]`.
We skip ahead and have `previous_value == None` and do one more iteration, i.e., we compare `data[i+1] - data[i]`.
Not sure if relevant, maybe double check the paper.